Linguini.Syntax/IO/ZeroCopyReader.cs

```diff
@@ -160,7 +161,8 @@ public ReadOnlyMemory<char> ReadSlice(int start, int end)
 
         public string ReadSliceToStr(int start, int end)
         {
-            return new(_unconsumedData.Slice(start, end - start).Span);
+            MemoryMarshal.TryGetString(_unconsumedData.Slice(start, end - start), out var text, out var _, out var _);
```

Comment:
This is incorrect. You want the `var text = MemoryMarshal.TryGetString` .
E.g. I have following string `abc`. I say `ReadSliceToStr(0,1)` and expect `a` the method as written returns `abc`.

I'd also put this into separate `#if net461 block` and keep original imlementation for brevity.